kubeform/pkg/provider/provider.go

```diff
@@ -166,7 +161,24 @@ func createKubeconfig(d *schema.ResourceData, meta interface{}) error {
 }
 
 func deleteKubeconfig(d *schema.ResourceData, meta interface{}) error {
+	cfg, err := kubeconfigInit(d, meta)
+	if err != nil {
+		return err
+	}
+
+	po := clientcmd.NewDefaultPathOptions()
+	if path, ok := d.GetOk("path"); ok {
+		po.LoadingRules.ExplicitPath = path.(string)
+	}
+
+	// Retry until modifyConfig succeeds or times out.
+	log.Printf("[DEBUG] updating kubeconfig")
+	if !poll(cfg.configPollInterval, cfg.ConfigPollTimeout, updateConfig(po, cfg.kubeConfig, removeConfig)) {
+		return fmt.Errorf("cluster components never turned healthy")
```

Comment:
This is just a case of error string copy/paste fail. I fixed this in the `createKubeconfig()` function which you reviewed earlier today and overlooked it here.

We do not check for cluster's health here, just for kubeconfig to update, i.e. delete the entry. If the operation to delete the kubeconfig entries fail, it must be reported to the user, so this is important.